telemetry/metrics.go

```diff
@@ -140,7 +135,61 @@ func (m *Metrics) Gather(format string) (GatherResponse, error) {
 	}
 }
 
-func (m *Metrics) gatherPrometheus() (GatherResponse, error) {
+// ModuleMeasureSince provides a short hand method for emitting a time measure
+// metric for a module with a given set of keys. If any global labels are defined,
+// they will be added to the module label.
+func (m *metrics) ModuleMeasureSince(module string, start time.Time, keys ...string) {
+	gometrics.MeasureSinceWithLabels(
+		keys,
+		start.UTC(),
+		append([]gometrics.Label{NewLabel(MetricLabelNameModule, module)}, m.cnf.globalLabels...),
+	)
+}
+
+// ModuleSetGauge provides a short hand method for emitting a gauge metric for a
+// module with a given set of keys. If any global labels are defined, they will
+// be added to the module label.
+func (m *metrics) ModuleSetGauge(module string, val float32, keys ...string) {
+	gometrics.SetGaugeWithLabels(
+		keys,
+		val,
+		append([]gometrics.Label{NewLabel(MetricLabelNameModule, module)}, m.cnf.globalLabels...),
+	)
+}
+
+// IncrCounter provides a wrapper functionality for emitting a counter metric with
+// global labels (if any).
+func (m *metrics) IncrCounter(val float32, keys ...string) {
+	gometrics.IncrCounterWithLabels(keys, val, m.cnf.globalLabels)
+}
+
+// IncrCounterWithLabels provides a wrapper functionality for emitting a counter
+// metric with global labels (if any) along with the provided labels.
+func (m *metrics) IncrCounterWithLabels(keys []string, val float32, labels []gometrics.Label) {
+	gometrics.IncrCounterWithLabels(keys, val, append(labels, m.cnf.globalLabels...))
+}
+
+// SetGauge provides a wrapper functionality for emitting a gauge metric with
+// global labels (if any).
+func (m *metrics) SetGauge(val float32, keys ...string) {
+	gometrics.SetGaugeWithLabels(keys, val, m.cnf.globalLabels)
+}
+
+// SetGaugeWithLabels provides a wrapper functionality for emitting a gauge
+// metric with global labels (if any) along with the provided labels.
+func (m *metrics) SetGaugeWithLabels(keys []string, val float32, labels []gometrics.Label) {
+	gometrics.SetGaugeWithLabels(keys, val, append(labels, m.cnf.globalLabels...))
+}
+
+// MeasureSince provides a wrapper functionality for emitting a a time measure
+// metric with global labels (if any).
+func (m *metrics) MeasureSince(start time.Time, keys ...string) {
+	gometrics.MeasureSinceWithLabels(keys, start.UTC(), m.cnf.globalLabels)
+}
+
+// ─── Utils ──────────────────────────────────────────────────────────────────────
```

Comment:
Nit: can we remove these types of comments please?

Reply:
removed.